.github/workflows/install-valkey/action.yml

```diff
@@ -19,30 +19,52 @@ inputs:
 
 env:
     CARGO_TERM_COLOR: always
-    VALKEY_MIN_VERSION: "7.2.5"
+    VALKEY_MIN_VERSION: "7.2"
 
 runs:
     using: "composite"
 
     steps:
         - name:
               Cache Valkey
-              # TODO: remove the musl ARM64 limitation when https://github.com/actions/runner/issues/801 is resolved
-          if: ${{ inputs.target != 'aarch64-unknown-linux-musl' }}
+              # TODO: self-hosted runners are actually cloning the repo, using the cache from the previous run
+              # will not work as expected. We need to find a way to cache the valkey repo on the runner itself.
+          if: ${{!contains(inputs.target, 'aarch64') }}
```

Comment:
Try using `inputs.target` in cache key name
Delete all caches before re-running the build

Reply:
So what is the goal of caching if i delete?

Reply:
It is better to use the cache to speed up CI tasks. Saving there 2 min will speed up nightly run for an hour.

Reply:
Miscommunication — are you talking about the self-hosted?
In the self-hosted we do “cache”, but there's no real caching like in the hosted since the resources are on our machine, so caching means nothing.
Let's say you pull the cloned repo and the built engine — the bin that's run is not the repo, it's what in one of the bin folders. To change it, you need to override what in the bin.
It's starting to become tricky — how do I know which bin? Is the symlink is pointing to ValKey-server bin (v8.0), or to Redis-server(v6.2)?
We can implement something here, of course, but it's a bit more than simply caching, since having repo and build cached don't change what will run.

The fastest way to do so is to have a few repos for each ver, all prebuilt, and change the alias per version run, and some logic to fetch pull and rebuild on version patching.
But, it's a separate PR for less close release.

I'll open an issue to remember.